tidy3d/components/medium.py

```diff
@@ -2927,6 +2927,51 @@ class CustomAnisotropicMediumInternal(CustomAnisotropicMedium):
     )
 
 
+class AbstractNonlinearMedium(AbstractMedium, ABC):
+    """Abstract nonlinear medium.
+
+    Note
+    ----
+    The nonlinear constitutive relation is solved iteratively; it may not converge
+    for large nonlinearities. Increasing `numiters` can help with convergence.
+
+    """
+
+    numiters: pd.PositiveInt = pd.Field(
+        1,
+        title="Number of iterations",
+        description="Number of iterations for solving nonlinear constitutive relation.",
+    )
+
+
+class KerrMedium(AbstractNonlinearMedium, Medium):
+    """Kerr medium described by a chi3 nonlinear susceptibility.
+
+    Note
+    ----
+    The instantaneous nonlinear polarization is given by
+    .. math::
+
+        P_{NL} = \\epsilon_0 \\chi_3 |E|^2 E
+
+    Note
+    ----
+    For complex fields (e.g. when using Bloch boundary conditions), the nonlinearity
+    is applied separately to the real and imaginary parts.
```

Comment:
could you be more explicit? so does it mean that the P_NL imaginary and real parts are computed separately from |real(E)|^3 and  |imag(E)|^3 respectively? or something else? what are the practical considerations of this? for example, is it still accurate to use KerrMedium with bloch boundaries?

Reply:
yeah, it's still accurate to use KerrMedium with bloch boundaries. It wouldn't be physical do use the complex fields directly in definining the instantaneous nonlinearity, since the instantaneous physical fields are real.